### PR TITLE
[General] Semantic release, contributing guide, templates

### PR DESCRIPTION
**Pull request by Krizzu**
As part of #21 cleanup, adding:

- Semantic release (releases from master)
- Contribution guide 
- Github templates

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+We'd like to make contributions as pleasent as possible, so here's a small guide of how we see it. Happy to hear your feedback about anything, so please let us know.
+
+
+## Tests
+We use `flow` for type check, `eslint` with `prettier` for linting/formatting, `jest/detox` for tests (unit and e2e). All tests are run on CircleCI for all opened pull requests, but you should use them locally when making changes.
+
+* `yarn test`: Run all tests, except for e2e (see note below).
+* `yarn test:lint`: Run `eslint` check.
+* `yarn test:flow`: Run `flow` typechecking.
+* `yarn test:e2e:<ios|android>`: Runs e2e tests. Before you can run it, you should build the app that can be run, by using `yarn build:e2e<ios|android>`.
```
Comment (matt-oakes) [inline CONTRIBUTING.md]: Add missing `:`

```suggestion
* `yarn test:e2e:<ios|android>`: Runs e2e tests. Before you can run it, you should build the app that can be run, by using `yarn build:e2e:<ios|android>`.
```

File: package.json [hunk 2]
```diff
@@ -28,14 +28,16 @@
     "test:flow": "flow check",
     "test:lint": "eslint 'lib/**/*.js' 'example/**/*.js'",
     "test:e2e:ios": "detox test -c ios",
-    "test:e2e:android": "detox test -c android"
+    "test:e2e:android": "detox test -c android",
```
Comment (matt-oakes) [inline package.json]: I like the `build:e2e:<platform>` and `test:e2e:<platform>` names. I'll update NetInfo to match this.

File: .circleci/config.yml [hunk 3]
```diff
@@ -304,4 +316,11 @@ workflows:
...
-            - "Build: Android release apk"
\ No newline at end of file
+            - "Build: Android release apk"
+      - "Publish: semantic release":
+          requires:
+            - "Test: Android e2e"
+            - "Test: iOS e2e"
+          filters:
+            branches:
+              only: master
```
Comment (matt-oakes) [inline .circleci/config.yml]: New line at end of file

```suggestion
              only: master

```

**Comment by Krizzu** [comment]
Thanks, @matt-oakes !

**Pull request merged**